astropy/coordinates/sky_coordinate.py

```diff
@@ -1850,79 +1850,70 @@ def radial_velocity_correction(
             raise ValueError(
                 "`kind` argument to radial_velocity_correction must "
                 f"be 'barycentric' or 'heliocentric', but got '{kind}'"
             )
 
+        pos_earth, v_origin_to_earth = get_body_barycentric_posvel("earth", obstime)
+        if kind == "heliocentric":
+            v_origin_to_earth -= get_body_barycentric_posvel("sun", obstime)[1]
+
         gcrs_p, gcrs_v = location.get_gcrs_posvel(obstime)
         # transforming to GCRS is not the correct thing to do here, since we don't want to
         # include aberration (or light deflection)? Instead, only apply parallax if necessary
         icrs_cart = coo_at_rv_obstime.icrs.cartesian
-        icrs_cart_novel = icrs_cart.without_differentials()
-        if self.data.__class__ is UnitSphericalRepresentation:
-            targcart = icrs_cart_novel
-        else:
+        targcart = icrs_cart.without_differentials()
+        if self.data.__class__ is not UnitSphericalRepresentation:
             # skycoord has distances so apply parallax
-            obs_icrs_cart = pos_earth + gcrs_p
-            targcart = icrs_cart_novel - obs_icrs_cart
+            targcart -= pos_earth + gcrs_p
             targcart /= targcart.norm()
 
-        if kind == "barycentric":
-            beta_obs = (v_origin_to_earth + gcrs_v) / speed_of_light
-            gamma_obs = 1 / np.sqrt(1 - beta_obs.norm() ** 2)
-            gr = location.gravitational_redshift(obstime)
-            # barycentric redshift according to eq 28 in Wright & Eastmann (2014),
-            # neglecting Shapiro delay and effects of the star's own motion
-            zb = gamma_obs * (1 + beta_obs.dot(targcart)) / (1 + gr / speed_of_light)
-            # try and get terms corresponding to stellar motion.
-            if icrs_cart.differentials:
-                try:
-                    ro = self.icrs.cartesian
-                    beta_star = ro.differentials["s"].to_cartesian() / speed_of_light
-                    # ICRS unit vector at coordinate epoch
-                    ro = ro.without_differentials()
-                    ro /= ro.norm()
-                    zb *= (1 + beta_star.dot(ro)) / (1 + beta_star.dot(targcart))
-                except u.UnitConversionError:
-                    warnings.warn(
-                        "SkyCoord contains some velocity information, but not enough to"
-                        " calculate the full space motion of the source, and so this"
-                        " has been ignored for the purposes of calculating the radial"
-                        " velocity correction. This can lead to errors on the order of"
-                        " metres/second.",
-                        AstropyUserWarning,
-                    )
-
-            zb = zb - 1
-            return zb * speed_of_light
-        else:
+        if kind == "heliocentric":
             # do a simpler correction ignoring time dilation and gravitational redshift
```

Comment:
Rephrase to `Do a simpler correction than for barycentric, ignoring ....` (since now the barycentric code has not yet been seen)

Reply:
Yes, I'll update the comment.